internal/mode/static/telemetry/collector.go

```diff
@@ -147,3 +158,36 @@ func collectGraphResourceCount(
 
 	return ngfResourceCounts, nil
 }
+
+func collectNGFReplicaCount(ctx context.Context, k8sClient client.Reader, podNSName types.NamespacedName) (int, error) {
+	var pod v1.Pod
+	if err := k8sClient.Get(ctx,
+		types.NamespacedName{Namespace: podNSName.Namespace, Name: podNSName.Name},
+		&pod,
+	); err != nil {
+		return 0, err
+	}
+
+	podOwnerRefs := pod.GetOwnerReferences()
+	if podOwnerRefs == nil {
+		return 0, fmt.Errorf("could not get owner reference of NGF Pod")
+	}
+	if len(podOwnerRefs) != 1 {
+		return 0, fmt.Errorf("multiple owner references of NGF Pod")
+	}
+
+	switch kind := podOwnerRefs[0].Kind; kind {
+	case "ReplicaSet":
+		var replicaSet appsv1.ReplicaSet
+		if err := k8sClient.Get(ctx,
```

Comment:
This function calls `k8sClient.Get(...)` twice. With the current way the collector tests work, I won't be able to correctly test this implementation as we cannot stub the `k8sClient.Get(...)` twice with different results. In addition, looking into the future when more data points are collected, we will most likely have many calls to `Get` or `List` from the same `k8sClient` for different collected resources. How do we handle this?

Reply:
I see a few options:

1. Extract lines 163-177 into a function that Gets the NGF Pod's owner ref. Then, each function only has one call to `k8sClient.Get`. 
2. Add logic to the `k8sClient.Get` stub so that it returns different results depending on the type of object passed in. 
3. Leverage the fake's `GetReturnsOnCall` method. This will allow you to set different return values based on the number of times the Get method has been called. So you can return nil for the first call the Get, and an error on the second.

Reply:
🤦  I completely missed option 2... I think thats the right approach.